plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreProcedure.java

```diff
@@ -813,7 +832,19 @@ public Collection<PostgrePrivilege> getPrivileges(DBRProgressMonitor monitor, bo
 
     @Override
     public DBSObject refreshObject(@NotNull DBRProgressMonitor monitor) throws DBException {
-        return getContainer().getProceduresCache().refreshObject(monitor, getContainer(), this);
+        boolean oldProcWasCreatedOrReplaced = this.procWasCreatedOrReplaced;
+        long oldOid = this.getObjectId();
+        long oldXmin = this.getXmin();
+        PostgreProcedure pgProc = getContainer().getProceduresCache().refreshObject(monitor, getContainer(), this);
+        if (oldProcWasCreatedOrReplaced) {
+            if ((oldOid != 0) && (oldXmin != 0) && (pgProc.oid == oldOid) && (pgProc.xmin == oldXmin)) {
+                DBWorkbench.getPlatformUI().showWarningMessageBox(PostgreSQLMessages.procedure_warning_name, 
+                    PostgreSQLMessages.procedure_warning_text);
+                this.setProcWasCreatedOrReplaced(false); // let throw warning once
+                pgProc.setProcWasCreatedOrReplaced(false);
+            }
+        }
```

Comment:
Too many redundant parentheses; nested `if`-s can be merged

```suggestion
        if (oldProcWasCreatedOrReplaced && oldOid != 0 && oldXmin != 0 && pgProc.oid == oldOid && pgProc.xmin == oldXmin) {
            DBWorkbench.getPlatformUI().showWarningMessageBox(PostgreSQLMessages.procedure_warning_name, 
                PostgreSQLMessages.procedure_warning_text);
            this.setProcWasCreatedOrReplaced(false); // let throw warning once
            pgProc.setProcWasCreatedOrReplaced(false);
        }
```